src/tools/rbd/action/Bench.cc

```diff
@@ -250,13 +250,16 @@ int do_bench(librbd::Image& image, io_type_t io_type,
 
   printf("  SEC       OPS   OPS/SEC   BYTES/SEC\n");
   uint64_t off;
+
   for (off = 0; off < io_bytes; ) {
-    b.wait_for(io_threads - 1);
     i = 0;
     while (i < io_threads && off < io_bytes) {
-      if (!b.start_io(io_threads, thread_offset[i], io_size, op_flags)) {
-        break;
+      b.wait_for(io_threads - 1);
+
+      while (!b.start_io(io_threads, thread_offset[i], io_size, op_flags)) {
```

Comment:
What's the purpose of this ```while``` loop? If you have already waited for a "thread" to become available for IO on line 257, IO should be able to start. In fact, you might as well remove the return ```bool``` from the ```start_io``` method since it should always be able to start.